File: app/src/main/java/com/example/neon/deardiary/diaryquery/DiaryQueryAdapter.java
package com.example.neon.deardiary.diaryquery;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.example.neon.deardiary.R;
import com.example.neon.deardiary.data.Diary;
import com.example.neon.deardiary.util.Strings;

import java.util.ArrayList;

class DiaryQueryAdapter extends BaseAdapter {
    private Context mContext;
    private ArrayList<Diary> mDiaryList;

    DiaryQueryAdapter(Context context) {
        this.mContext = context;
        this.mDiaryList = new ArrayList<>();
    }

    public void setData(ArrayList<Diary> data) {
        mDiaryList = data;
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        return this.mDiaryList.size();
    }

    @Override
    public Object getItem(int position) {
        return mDiaryList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder holder;
        Diary diary = mDiaryList.get(position);
        if (convertView == null) {
            LayoutInflater inflater = LayoutInflater.from(mContext);
            convertView = inflater.inflate(R.layout.item_search_list, null);
            holder = new ViewHolder();
            holder.mContentTV = (TextView) convertView.findViewById(R.id.result_content);
            holder.mDateTV = (TextView) convertView.findViewById(R.id.result_date);
            holder.mTitleTV = (TextView) convertView.findViewById(R.id.result_title);
            convertView.setTag(holder);
        }
        holder = (ViewHolder) convertView.getTag();

        String year = diary.getYear() + "";
        String month = diary.getMonth() + "";
        String dayOfMonth = diary.getDayOfMonth() + "";
        String hour = Strings.formatNumber(diary.getHour());
        String minute = Strings.formatNumber(diary.getMinute());
        String time = year + "年" + month + "月" + dayOfMonth + "日 "
                + hour + "时" + minute + "分";
        holder.mTitleTV.setText(diary.getTitle());
        holder.mContentTV.setText(diary.getContent());
        holder.mDateTV.setText(time);

        return convertView;
    }

    private class ViewHolder {
        private TextView mDateTV;
        private TextView mTitleTV;
        private TextView mContentTV;
    }


}

File: app/src/main/java/com/example/neon/deardiary/diaryquery/DiaryQueryAdapter.kt
package com.example.neon.deardiary.diaryquery

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.example.neon.deardiary.R
import com.example.neon.deardiary.data.Diary
import com.example.neon.deardiary.diaryedit.DiaryEditActivity
import com.example.neon.deardiary.utils.StringUtil
import java.util.ArrayList

internal class DiaryQueryAdapter(private val mContext: Context) : RecyclerView.Adapter<DiaryQueryAdapter.ViewHolder>() {
    private var mDiaryList: ArrayList<Diary> = ArrayList()

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder {
        val itemView = LayoutInflater.from(mContext).inflate(R.layout.item_search_list, null)
        return ViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: ViewHolder?, position: Int) {
        val diary = mDiaryList[position]
        val year = diary.year.toString()
        val month = diary.month.toString()
        val dayOfMonth = diary.dayOfMonth.toString()
        val hour = StringUtil.formatNumber(diary.hour)
        val minute = StringUtil.formatNumber(diary.minute)
        val time = "$year 年$month 月$dayOfMonth 日 $hour 时$minute 分"
        holder !!.mTitleTV.text = diary.title
        holder.mContentTV.text = diary.content
        holder.mDateTV.text = time
        holder.itemView.setOnClickListener {
            val bundle = Bundle()
            bundle.putSerializable(DiaryEditActivity.DATA_DIARY, diary)
            val intent = Intent(mContext, DiaryEditActivity::class.java)
            intent.putExtras(bundle)
            mContext.startActivity(intent)
        }
    }

    override fun getItemCount(): Int = mDiaryList.size

    fun setData(data: ArrayList<Diary>) {
        mDiaryList = data
        notifyDataSetChanged()
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var mDateTV: TextView = itemView.findViewById(R.id.result_date) as TextView
        var mTitleTV: TextView = itemView.findViewById(R.id.result_title) as TextView
        var mContentTV: TextView = itemView.findViewById(R.id.result_content) as TextView
    }
}
